### ISSUE TITLE
Running :GutentagsUpdate makes most of the screen black

### ISSUE THREAD

**Issue by darthdeus** [ludovicchabant/vim-gutentags#123]
I'm not really sure how to describe this the best, but if I run `:GutentagsUpdate` on any file within a git repo, it turns from this

![](http://i.imgur.com/Z0fdx6v.png)

to this

![](http://i.imgur.com/jyXCAEh.png)

If I move the cursor around it redraws the lines under the cursor properly

![](http://i.imgur.com/G9V823A.png)

and doing `:redraw!` fixes everything.

Any idea what could be causing this? Is it an issue with my `~/.vimrc`, or with gutentags? This never happened to me with other vim plugins.

This is on VIM 8.0 (2016 Sep 12, compiled Mar 6 2017 14:11:23) - the latest version on Arch Linux. I've tried both inside tmux and outside and in two different terminals.

**Comment by ludovicchabant** [comment]
Hi!

You're not the only one to report a redraw issue with Gutentags, but I have absolutely no idea what could cause it, and I haven't been able to reproduce it locally :(

The only workaround I can offer right now, at least until someone has an idea or a lead, is to locally setup the `GutentagsUpdated` User auto-command, which is executed after an update. You can make it call `:redraw`.

**Comment by darthdeus** [comment]
Thanks, I'll see if that helps.

@ludovicchabant Though I just diagnosed a new issue with gutentags, specifically https://github.com/junegunn/fzf.vim/issues/344 (I'm about 99% sure it was caused by gutentags).

TL;DR: _sometimes_ the `:pwd` changes to a home directory. While this happened when using FZF, the issue only started after I added gutentags to my vimrc and it stopped happening once I removed gutentags a few days later.

**Comment by ludovicchabant** [comment]
Gutentags does indeed set the current working directory, but that's only for generating tags -- it then restores the previous directory. There may be a bug that makes it not restore it correctly sometimes though. I'm reducing the amount of `chdir` calls right now to see if that would fix it.

As for the redrawing issues, I'm still in the dark (ho ho) about it.

**Comment by mgedmin** [comment]
I'm pretty sure this is what happens when you do `:silent !any-shell-command` in (recentish) vim.  I'm pretty sure this is a bug in vim (it clears the screen before running the shell command but then doesn't redraw it when the command returns), but I was unable to convince vim's maintainers of this.

See https://github.com/vim/vim/issues/1253.

**Comment by battlesnake** [comment]
Running vim 8.0 inside tmux 2.5-3 in xterm on Arch, with `TERM=screen-256color`, I get that issue if I run GutentagsUpdate directly.

For now, I use:

    nmap <F4> :GutentagsUpdate<CR><C-l>

**Comment by ludovicchabant** [comment]
Super belated thanks @battlesnake for the investigation -- I haven't heard back on this anyway and it seems a bit out of Gutentags' hands, so I'll close it.
